File: 프로그래머스/1/12982. 예산/예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> d, int budget) {
    int answer = 0;
    
    sort(d.begin(), d.end());
    int cur_cost = 0;
    
    for(int i = 0 ; i < d.size() ; i++){
        
       if(cur_cost + d[i] <= budget){
           cur_cost += d[i];
           answer++;
           continue;
       } else{
           break;
       }
        
    }
    
    
    return answer;
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        Arrays.sort(d);
        
        int total = 0;
        int cnt = 0;
        
        for(int b : d){
            if(total + b > budget)
                break;
            
            total += b;
            cnt++;
        }
        
        return cnt;
    }
}